crates/crypto/src/hashes/sha256.rs

```diff
@@ -10,22 +10,23 @@ pub fn sha256(data: &[u8]) -> [u8; 32] {
         use ring::digest;
         let hash = digest::digest(&digest::SHA256, data);
 
-        // we're pretty sure this will always be 32 bytes long
-        assert!(
-            hash.as_ref().len() == digest::SHA256.output_len,
-            "ring's Sha256 implementation has returned a digest of len {}, expected 32",
-            hash.as_ref().len()
-        );
-
-        hash.as_ref().try_into().unwrap()
+        // Convert the digest to a fixed-size array
+        let mut array = [0u8; 32];
+        array.copy_from_slice(hash.as_ref());
+        array
```

Comment:
Double-check [the documentation](https://doc.rust-lang.org/std/primitive.slice.html#panics-30): `copy_from_slice` panics in exactly the same situation as `try_into().unwrap()`. We don't need error handling here, there is an implicit code contract that the `ring` implementation of SHA256 will always return 32 bytes. This `assert!` here is just as a sanity check and gives us a more explicit reason for a panic if this contract is ever broken.